composer.json

```diff
@@ -15,6 +15,7 @@
     "minimum-stability": "dev",
     "require": {
         "php": "^7.1",
+        "ext-dom": "*",
```

Comment:
I have a question: Isn't this extension optional? The usage of XML in Doctrine is optional too for a user, isn't it?

Reply:
Technically, it's an optional dependency. However, making it an explicit dependency is the defensive way here: we prevent the code from blowing up in the users face with a fatal error if they use XML mappings.

Remember that other bundles (e.g. the FOSUserBundle) specify XML mappings which are automatically imported, so the user won't know that he's using XML mappings and thus should add the `ext-dom` dependency.

One more thought: moving this to `suggest` would require all bundles that specify XML mappings to add a `require` entry for `ext-dom`, which would be a BC break towards the bundles. IMO it's safer to add it as an explicit dependency.

Reply:
Talking about "Bundles" sounds more like Symfony instead of Doctrine. This repository is independent from frameworks. Should in this case the `require` be added in DoctrineBundle? What are "all bundles"?

Reply:
@SenseException replace with "bundles" with "anything that provides XML mappings". Suggesting `ext-dom` instead of requiring it will break third party software if they already use XML mappings. That's a BC break that would move this feature back to 3.0. The way it is now it's possible to add this to the 2.x line.

Reply:
@alcaeus I rebased on 3.x yesterday :sweat_smile:

Reply:
Oh, didn't see that, my bad. IMO it's better to have the explicit dependency anyways. 👍

Reply:
I have mixed feelings about this. One one hand, it's true that user would need explicit ext-dom dependency. On the other one, we should check at runtime for DOM presence, it just feels wrong to force people with annotations install DOM. (Well, we can say the same about doctrine/annotations, but that's slightly different story).
Since this PR is now 3.0 only, it should be fine fine to do either way, but question is which one is better.

Reply:
> On the other one, we should check at runtime for DOM presence, it just feels wrong to force people with annotations install DOM.

https://webmozart.io/blog/2014/10/24/defining-php-annotations-in-xml/